src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -122,37 +126,63 @@ public static void finalizeRegion(final AssemblyRegion region,
         final byte minTailQualityToUse = errorCorrectReads ? HaplotypeCallerEngine.MIN_TAIL_QUALITY_WITH_ERROR_CORRECTION : minTailQuality;
 
         final List<GATKRead> readsToUse = new ArrayList<>();
+        final List<GATKRead> hardClippedReadsToUse = new ArrayList<>();
```

Comment:
what cases would trackHardclippedReads be true?

Reply:
Honestly? the only place we ever are tracking these is the pileup code (for right now... I can think of some places where it might be useful to get rid of the softclips for downstream processing). I actually am not 100% certain its important for this codepath as it is though since we are now discounting the read kmers overlapping softclips in the fitlering...